### PR TITLE
Adding hostipc policy for new scc

### PR DESCRIPTION
**Pull request by igorkchyts**
Adding hostipc policy that ensures that created scc are not allowing access to the host PID and IPC namespaces

### CODE REVIEW

File: kyverno/authorization/disallow-host-ipc.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    - name: validate-hostIPC-scc
+      match:
+        resources:
+          kinds:
+            - SecurityContextConstraints
+      validate:
+        message: "Use of host PID ands IPC namespaces is not allowed"
+        pattern:
+          =(allowHostPID): "false"
+          =(allowHostIPC): "false"
```
Comment (michaelkotelnikov) [inline kyverno/authorization/disallow-host-ipc.yaml:22]: Exclude the SCCs that allow HostIPC by default.

File: kyverno/authorization/disallow-host-ipc.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - name: validate-hostIPC-pod
+      match:
+        resources:
+          kinds:
+            - Pod
+      validate:
+        message: "Use of host PID ands IPC namespaces is not allowed"
```
Comment (michaelkotelnikov) [inline kyverno/authorization/disallow-host-ipc.yaml:29]: Provide a unique message for violations caused by editing `pod` resources

File: kyverno/authorization/disallow-host-ipc.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+spec:
+  validationFailureAction: enforce
+  rules:
+    - name: validate-hostIPC-scc
+      match:
+        resources:
+          kinds:
+            - SecurityContextConstraints
+      validate:
+        message: "Use of host PID ands IPC namespaces is not allowed"
```
Comment (michaelkotelnikov) [inline kyverno/authorization/disallow-host-ipc.yaml:19]: provide a unique message for violations caused by editing `ssc` resources.

File: kyverno/README.md [hunk 4]
```diff
@@ -24,6 +24,7 @@ No policies yet       |  |
 Policy  | Description | Prerequisites
 ------- | ----------- | -------------
 [protect-default-scc](./authorization/protect-default-scc.yaml) | Ensures that default scc are not being modified or deleted |
+[disallow-host-ipc-scc](./authorization/disallow-host-ipc-scc.yaml) | Ensures that access to the host PID and IPC namespaces is restricted |
```
Comment (michaelkotelnikov) [inline kyverno/README.md:27]: Provide more information tothe description.

**Review by michaelkotelnikov — CHANGES_REQUESTED** [review]

File: kyverno/authorization/host-namespaces/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Host Namespaces policies
+
+Policies placed under directory 'host-namespaces' targets to disable pods access to sensitive host's resources.
+
+Access to several host namespaces (Process ID namespaces, Inter-Process Communication namespaces, and network namespaces) allow access to shared information and can be used to elevate privileges.
```
Comment (michaelkotelnikov) [inline kyverno/authorization/host-namespaces/README.md:5]: Change `allow` to `allows`. @igorkchyts

**Review by michaelkotelnikov — CHANGES_REQUESTED** [review]

**Review by michaelkotelnikov — CHANGES_REQUESTED** [review]

File: kyverno/authorization/host-namespaces/disallow-host-ipc.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  validationFailureAction: enforce
+  rules:
+    - name: validate-hostIPC-scc
+      match:
+        resources:
+          kinds:
+            - SecurityContextConstraints
+      exclude:
+        resources:
+          name: hostaccess
```
[the remark this answers is not in the document]
Comment (igorkchyts) [reply kyverno/authorization/host-namespaces/disallow-host-ipc.yaml:20]: @michaelkotelnikov Resolved by adding 'exclude.resource.name' for each SCC
Please validate

[the remark this answers is not in the document]
Comment (igorkchyts) [reply kyverno/authorization/host-namespaces/disallow-host-ipc.yaml:20]: @michaelkotelnikov 
The rule works as expected for 'Pod' resource types. 
After playing around for a while I keep getting ignored when excluding "SecurityContextConstaints". 
Open new issue in kyverno :
https://github.com/kyverno/kyverno/issues/1836

**Review by michaelkotelnikov — APPROVED** [review]
@igorkchyts Looks good to me now.

**Pull request merged**